sql/core/src/main/scala/org/apache/spark/sql/ForeachWriter.scala

```diff
@@ -68,9 +71,9 @@ import org.apache.spark.sql.streaming.ContinuousQuery
 abstract class ForeachWriter[T] extends Serializable {
 
   /**
-   * Called when starting to process one partition of new data in the executor side. `version` is
-   * for data deduplication. When recovering from a failure, some data may be processed twice. But
-   * it's guarantee that they will be opened with the same "version".
+   * Called when starting to process one partition of new data in the executor. The `version` is
+   * for data deduplication when there are failures. When recovering from a failure, some data may
+   * be generated multiple times but they will always have the same version.
    *
    * If this method finds this is a partition from a duplicated data set, it can return `false` to
```

Comment:
nit: finds using the `partitionId` and `version` that this partition has already been processed, it can return ...